### ISSUE TITLE
Git for Windows 2.x 64bit support

### ISSUE THREAD

**Issue by hantsy** [docker/toolbox#95]
I have installed the [latest Git for windows 2.x](https://github.com/git-for-windows/git/releases), it includes a 64bit version.

But it can not be detected by Toolbox at all. It seems Toolbox requires x86 version, and it will cause failure during installation.

I am using Windows 10. I have tried Docker Toolbox 1.8.0.a and 1.8.1.

**Comment by southwolf** [comment]
Seems Docker Toolbox adds tools to original git ("msys tools"?). My git 2.5 path settings override git 1.x bundled in Docker Toolbox, when I run docker terminal, it says "bash not found".

**Comment by djsylvester** [comment]
@hantsy Win 7x64 in my case but it did install. It just won't start.

@southwolf I don't think so. First time I installed I already had git x86 installed so unchecked the box in the toolbox installer. Everything worked fine.

I've been waiting a long time for the actual release of git x64 for Windows so when it came out I grabbed it. If you read the release notes you'll see that since v2.4.0:

The installer will now remove files from a previous Git for Windows versions, particularly important for 32-bit -> 64-bit upgrades

This is what I did:
1. uninstalled both x86 git, toolbox, and virtualbox. 2. Removed all directories and cleaned my registry of cruft.
3. Installed x64 git.
4. Installed toolbox while unchecking the install of git (don't want both).

I kept a log file of the entire install and see that it actually knows I have git in c:\program files (x64) and the installation completed fine, but I get a failure on quickstart :

Error: Unknown shell
bash: On: command not found
Looks like something went wrong... Press any key to continue...

I tried a few changes to the shell and bat scripts but had no luck.

**Comment by southwolf** [comment]
@djsylvester  Yes I got the same error. I assume Docker Toolbox adds extra tools to the original Git for Windows, or makes some special tricks on environment settings, so the toolbox cannot run with vanilla git.
 
```bash
trap '[ "$?" -eq 0 ] || read -p "Looks like something went wrong... Press any key to continue..."' EXIT
```

Looks like `[ "$?" -eq 0 ]` causes the failure

**Comment by hantsy** [comment]
My before failure occurred in the upgrading boot2docker stage.

Now I installed the latest 1.8.1b, which indicated it is compatible with Windows 10.

The **Toolbox QuickStart Terminal** link failed and it reported **bash not found**.

Finally, I ran `docker-machine` in cmd successfully when add **--shell cmd** parameter.

    docker-machine env --shell cmd  default

And the `docker` command worked, when I have set the variable environment printed in above command. 

    docker run hello-world

Ah, it worked!!!

**Comment by djsylvester** [comment]
@southwolf The <code>[ "$?" -eq 0 ]</code> doesn't <em>cause</em> a failure. That's just good scripting, checking the exit values of the previous command/loop/pipeline/etc. <code>"$?"</code> is returning 127 in this instance.

Also, running a hash against vanilla vs DTB install of git doesn't support the idea of <em>adding</em> anything extra either. It doesn't. It all has to do with the environment vars not being set. Fix below.

@hantsy Yeah, I did that a couple of times and it does work but Windows command prompt is just way to awful an experience, and Powershell isn't much better. I use Unix/Linux command line for almost everything locally (cygwin) and remotely. It's nice to have a single shell environment when you do a ton of stuff at the prompt.

So, this is how I fixed'er up so you can just use the Quickstart shortcut directly. I built on grabbing those env variables AND getting cmd out of the way.

1. Windows git doesn't create a .bash_profile or .bashrc since it's a relatively trimmed down version of the Bash. Create a .bash_profile and .bashrc in your home directory (I'm using a bash term here).
<code> touch .bash_profile .bashrc</code>
Yes, you can get away with just a bash_profile and skipping step 2 but my old habits won't die. When I have a mod to make my fingers go to to rc files, not profiles.

2. Add the this line to the .bash_profile
<code>source ~/.bashrc</code>

3. Add the lines from your <code>docker-machine env --shell cmd  default</code> being careful to convert the <code>DOCKER_CERT_PATH</code> for use with bash (<code>~/.docker/machine/machines/default</code>)

4. Edit start.sh and comment out:
<code>#eval docker-machine env default</code> and add:
<code>$DOCKER_MACHINE env --shell bash $VM</code>

That's it. Now you don't have to fool with cmd or pwrshell.

This also requires that you make sure your path includes both directories for Git and DTB. (DTB places its path in your "user" path, not global path btw).

One last thing. Git bash uses mintty for the terminal so you can also create a <code>.minttyrc</code> file to do some interesting things with the terminal that may be helpful. Plenty of man pages out there.

**Comment by jeffdm** [comment]
Thanks for the issue! The next version of Toolbox will include Git 2.5 and includes a fix to make this work.

**Comment by jeffdm** [comment]
This should be fixed in 1.8.1c for Windows. Thank again!
